scripts/upload.py

```diff
@@ -30,14 +31,33 @@ def upload(globpath, container, queue, sas_token_env, storage_account_uri):
 
             blob_client = BlobClient(account_url=storage_account_uri.format('blob'), container_name=container, blob_name=blob_name, credential=sas_token)
             
-            with open(infile, "rb") as data:
-                blob_client.upload_blob(data, blob_type="BlockBlob", content_settings=ContentSettings(content_type="application/json"))
-
-            if queue is not None:
-                queue_client = QueueClient(account_url=storage_account_uri.format('queue'), queue_name=queue, credential=sas_token, message_encode_policy=TextBase64EncodePolicy())
-                queue_client.send_message(blob_client.url)
-
-            getLogger().info("upload complete")
+            uploadedSuccessfully = False
+            for i in range(3):
```

Comment:
I would probably create a helper "retry" function that would take lambda and have all the looping and catching encapsulated there.

Reply:
Did this, give the new changes a look when you get a chance please. 👍

Reply:
Love the `retry_on_exception`!